cpp/src/arrow/integration/meson.build

```diff
@@ -20,8 +20,13 @@ install_headers(['json_integration.h'])
 exc = executable(
     'arrow-json-integration-test',
     sources: ['json_integration_test.cc'],
-    dependencies: [arrow_dep, rapidjson_dep, gflags_dep, gtest_dep],
-    link_with: [arrow_test_lib],
+    dependencies: [
+        arrow_dep,
+        arrow_testing_dep,
+        rapidjson_dep,
+        gflags_dep,
+        gtest_dep,
```

Comment:
Do we need `gtest_dep` with `arrow_testing_dep`?
`arrow_testing_dep` includes `gtest_main_dep`.

Reply:
Its a bit confusing and there might be a better way to word things, but with this PR we now have a few dependencies being declared:

 - arrow_testing_dep, which just exposes the testing library
 - arrow_test_dep, which exposes multiple dependencies that are used to compile and run tests in arrow
 - arrow_test_dep_no_main which is used by tests that declare their own main function

I could use `arrow_test_dep_no_main` here instead; it would add a few superfluous dependencies but is closer to what this test requires

Reply:
Ah, sorry. I was wrong.

Let's use `arrow_test_dep_no_main` here.